ludwig/schema/hyperopt/executor.py

```diff
@@ -0,0 +1,63 @@
+from dataclasses import field
+from typing import Dict
+
+from marshmallow import fields, ValidationError
+from marshmallow_dataclass import dataclass
+
+from ludwig.schema import utils as schema_utils
+from ludwig.schema.hyperopt.scheduler import SchedulerConfig, SchedulerDataclassField
+
+
+@dataclass
+class ExecutorConfig(schema_utils.BaseMarshmallowConfig):
+    """Basic executor settings."""
+
+    type: str = schema_utils.StringOptions(options=["ray"], default="ray", allow_none=False)
+
+    num_samples: int = schema_utils.PositiveInteger(default=10, description="")
+
+    time_budget_s: int = schema_utils.PositiveInteger(default=3600, description="")
+
+    cpu_resources_per_trial: int = schema_utils.PositiveInteger(default=1, description="")
+
+    gpu_resources_per_trial: int = schema_utils.PositiveInteger(default=1, description="")
+
+    scheduler: SchedulerConfig = SchedulerDataclassField(description="")
+
```

Comment:
I'd also add `kubernetes_namespace` (https://ludwig-ai.github.io/ludwig-docs/0.5/configuration/hyperparameter_optimization/#scheduler)

Reply:
added - is `None` the correct default?

Reply:
Yup